src/lib.rs

```diff
@@ -323,11 +324,11 @@ impl<K: Hash + Eq, V, S: HashState> LinkedHashMap<K, V, S> {
     /// assert_eq!(map.get(&2), Some(&20));
     /// ```
     #[inline]
-    pub fn pop_front(&mut self) -> Option<V> {
+    pub fn pop_front(&mut self) -> Option<(K, V)> {
         if self.len() > 0 {
             let lru = unsafe { (*self.head).prev };
             self.detach(lru);
-            return self.map.remove(&KeyRef{k: unsafe { &(*lru).key }}).map(|e| e.value)
+            return self.map.remove(&KeyRef{k: unsafe { &(*lru).key }}).map(|box e| (e.key, e.value))
```

Comment:
In order to avoid adding the `box_patterns` feature, can you make this something like:
```rust
return self.map.remove(&KeyRef{k: unsafe { &(*lru).key }}).map(|e| { let e = *e; (e.key, e.value) })
```